node/cn/handler.go

```diff
@@ -465,14 +465,24 @@ func (pm *ProtocolManager) handle(p Peer) error {
 }
 
 func (pm *ProtocolManager) processMsg(msgCh <-chan p2p.Msg, p Peer, addr common.Address, errCh chan<- error) {
-	for msg := range msgCh {
-		if err := pm.handleMsg(p, addr, msg); err != nil {
-			p.GetP2PPeer().Log().Error("ProtocolManager failed to handle message", "msg", msg, "err", err)
-			errCh <- err
-			return
+	_, fakeF := pm.fetcher.(*fetcher.FakeFetcher)
+	_, fakeD := pm.downloader.(*downloader.FakeDownloader)
+	if fakeD || fakeF {
```

Comment:
If fetcher or downloader is fake, then is it okay with ignoring all the other messages such as `NewBlockHashes`, `NewBlockMsg`?

Reply:
I think it's okay. Could you tell me what you are considering?

Reply:
Unhandling those messages will not update td for remote peers. Never mind.
BTW, is there any chance to turn on only one of fake fetcher/downloader? Otherwise, how about changing `||` to `&&`?

Reply:
As you said, both of fake modules will be used together in all valid scenarios. However, I wanna warn the node operators using one of fake modules and make them avoid unexpected failures because the received p2p messages maybe not handled properly.

Reply:
Understood. Thanks.